src/basic_recipes/barplot.jl

```diff
@@ -46,15 +49,60 @@ function AbstractPlotting.plot!(p::BarPlot)
         end
     end
 
-    bars = lift(p[1], p.fillto, p.width, in_y_direction) do xy, fillto, width, in_y_direction
+    bars = lift(p[1], p.fillto, p.width, p.dodge, p.x_gap, p.dodge_gap, p.stack, in_y_direction) do xy, fillto, width, dodge, x_gap, dodge_gap, stack, in_y_direction
+      
+        #
+        if dodge === automatic
+            n_dodge = 1
+        else
+            n_dodge = length(unique(dodge))
+        end
```

Comment:
Maybe `i_dodge` could be computed before this code block. Then this could simply be `n_dodge = maximum(i_dodge)`? That way if we change the logic to compute `i_dodge`, we only have to change it once.